corehq/apps/reports/util.py

```diff
@@ -442,11 +442,18 @@ def get_installed_custom_modules():
 
 
 def is_mobile_worker_with_report_access(couch_user, domain):
-    return (
-        couch_user.is_commcare_user
-        and domain is not None
-        and Domain.get_by_name(domain).default_mobile_worker_redirect == 'reports'
-    )
+    if not couch_user.is_commcare_user():
+        return False
+
+    if domain is not None:
+        role = couch_user.get_role(domain)
```

Comment:
I think this might be able to just return `couch_user.can_view_any_reports(domain)`. that logic looks more complicated but I think it works and might be nicer to keep everything in one place.